cmd/zfs/zfs_main.c

```diff
@@ -3746,8 +3746,13 @@ zfs_do_redact(int argc, char **argv)
 		    "specified\n"));
 		break;
 	case EINVAL:
-		(void) fprintf(stderr, gettext("redaction snapshot must be "
-		    "descendent of snapshot being redacted\n"));
+		if (strchr(bookname, '#') != NULL)
+			(void) fprintf(stderr, gettext(
+			    "redaction bookmark name must not contain '#'\n"));
+		else
+			(void) fprintf(stderr, gettext(
+			    "redaction snapshot must be descendent of"
+			    " snapshot being redacted\n"));
```

Comment:
nit: normally when breaking a line like this we add the space to the end of the previous line, in this case after "of" on line 3755